CMakeLists.txt

```diff
@@ -81,21 +83,22 @@ endif()
 
 if(NOT CUKE_DISABLE_GTEST)
     set(GTEST_USE_STATIC_LIBS ${CUKE_USE_STATIC_GTEST})
-    set(GMOCK_ROOT "${CMAKE_CURRENT_BINARY_DIR}/gmock")
+    if(NOT GMOCK_ROOT)
+        set(GMOCK_ROOT "${CMAKE_CURRENT_BINARY_DIR}/gmock")
+    endif()
     find_package(GMock REQUIRED)
 
     if(GTEST_FOUND)
         set(CUKE_GTEST_LIBRARIES
-            ${GTEST_BOTH_LIBRARIES}
-            ${CMAKE_THREAD_LIBS_INIT}
-        )
+            GTest::GTest
+            )
     endif()
     if(GMOCK_FOUND)
         set(CUKE_GMOCK_LIBRARIES
-            ${GTEST_BOTH_LIBRARIES}
-            ${GMOCK_BOTH_LIBRARIES}
-            ${CMAKE_THREAD_LIBS_INIT}
-        )
+            GTest::GTest
+            GMock::GMock
+            GMock::Main
+            )
```

Comment:
I did something like that a while back as an experiment: https://github.com/muggenhor/cucumber-cpp/compare/master...link-depends-only, and it's big enough a change that it definitely belongs in a different PR. But it does depend on doing the same kind of change that you're doing happening first.